core/templates/dev/head/i18n.js

```diff
@@ -62,16 +62,6 @@ oppia.controller('I18nFooter', [
         $http.put(siteLanguageUrl, {
           site_language_code: $scope.currentLanguageCode
         });
-      } else {
-        // $translate.use() sets a cookie for the translation language, but does
-        // so using the page's base URL as the cookie path. However, the base
-        // URL is modified in pages like /library, thus causing the cookie path
-        // to change; in such cases, the user's preferences are not picked up by
-        // other pages. To avoid this, we manually set the cookie using the '/'
-        // path each time a non-logged-in user changes their site language.
-        $cookies.put(
```

Comment:
With the removal of this clause, what is the behavior for a user who is not logged in and changes the translations?

Reply:
@seanlip, this issue (#4756) was for the guest user (i.e, who is not logged in) and for logged in user we use to save this changes in the backend and this issue resolves guest users translation issue.
Why I removed this:
This was of no use in the first place as this sets the cookie's data on root path but the get request happens with base path.

Reply:
OK, but what I'm asking is, let's say we have a guest user. They view the page in English and change the translations to Hindi. They then navigate around and refresh the page (or open Oppia in a different tab). What happens (and what ought to happen)?

Similar question for what if they close the browser and open it again?

Reply:
When a guest user will change the site language from the library footer this will get reflected to entire oppia.org using the cookies (with the session as expiry age) till they close the browser (i.e, same behavior if they will refresh the page or open Oppia in a different tab.) 
> Similar question for what if they close the browser and open it again?

If they close the browser the cookies will get deleted/cleared and when they will reopen the browser or oppia.org then the default (english) or [determinePreferredLanguage()](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/i18n.js#L102) will get selected.

Reply:
Ah, I see, $translate.use takes care of that. OK, makes sense, thanks for explaining!